src/canvas/SelectableCanvas.ts

```diff
@@ -1105,6 +1105,13 @@ export class SelectableCanvas<EventSpec extends CanvasEvents = CanvasEvents>
     }
     this._activeObject = object;
 
+    if (object instanceof ActiveSelection && this._activeSelection !== object) {
+      this._activeSelection.dispose();
```

Comment:
this is a later addition that is problematic.
Disposing an activeSelection means disposing contained objects.
We can't do that. This is just setActiveObject() and we shouldn't dispose inside it  because all the resources the developer is using may be still completely valid and needed for his/her app.

The old active selection being dereferenced will be garbage collected at some point.